kubernetes/kubernetes.go

```diff
@@ -165,7 +173,7 @@ func CollectKubeMetrics(config KubeAgentConfig) {
 		log.Warnf("Warning: Non-fatal error occurred retrieving baseline metrics: %s", err)
 	}
 
-	err = performConnectionChecks(&kubeAgent)
+	err = performConnectionChecks(&kubeAgent, customS3Mode)
```

Comment:
Would make more sense imo to just not call this method in `customS3Mode`

Reply:
I only did it this way to prevent gocyclo complaints. It is one if away from going over. 
I can just add //nolint